### PR TITLE
Update attestation plugin API and design to reflect implementation

### PR DESCRIPTION
**Pull request by shnwc**
Update attestation plugin API and design to reflect implementation, which puts header wrapping / processing inside plugins.

### CODE REVIEW

File: docs/DesignDocs/Attestation_API_Proposal.md [hunk 1]
```diff
@@ -591,11 +609,26 @@ by `oe_get_report()`.
...
-For evidence data generated by `oe_get_evidence()`, the list of claims returned
+For evidence data wrapped by a header with a format ID, the list of claims returned
 by `oe_verify_evidence()` will contain one claim named
-`OE_CLAIM_CUSTOM_CLAIMS_ID` if the evidence data contains a non-empty custom
-claims buffer. For an SGX ECDSA report or a raw SGX ECDSA quote,
-the list of claims will contain one claim named `OE_CLAIM_SGX_REPORT_DATA_ID`.
+`OE_CLAIM_CUSTOM_CLAIMS` if the evidence data contains a non-empty custom
+claims buffer. For evidence data without an embedded format ID,
+e.g. an SGX ECDSA report or a raw SGX ECDSA quote,
+the list of claims will contain one claim named `OE_CLAIM_SGX_REPORT_DATA`.
```
Comment (dthaler) [inline docs/DesignDocs/Attestation_API_Proposal.md:617]: The above statement would not be true for non-SGX TEEs, so needs to be reworded.
```suggestion
claims buffer. For evidence data containing an SGX ECDSA report or a raw SGX ECDSA quote without an embedded format ID,
the list of claims will contain one claim named `OE_CLAIM_SGX_REPORT_DATA`.
```

File: docs/DesignDocs/Attestation_API_Proposal.md [hunk 2]
```diff
@@ -491,7 +509,19 @@ by the verifier plugin entry points defined in the `oe_verifier_t` structure:
     point.
 - Verify evidence, optionally using a set of input endorsements and policies.
   - Entry point `oe_result_t (*verify_evidence)(oe_verifier_t* context, const uint8_t* evidence_buffer, size_t evidence_buffer_size, const uint8_t* endorsements_buffer, size_t endorsements_buffer_size, const oe_policy_t* policies, size_t policies_size, oe_claim_t** claims, size_t* claims_length)`.
-  - This entry point has the same definition as in the OE SDK
+  - `evidence_buffer` and `endorsements_buffer` contain the all evidence and
```
Comment (dthaler) [inline docs/DesignDocs/Attestation_API_Proposal.md:512]:
```suggestion
  - `evidence_buffer` and `endorsements_buffer` contain all the evidence and
```

Comment (shnwc) [reply docs/DesignDocs/Attestation_API_Proposal.md]: Thanks! Updated.

**Comment by shnwc** [comment]
bors try

**Pull request merged**